search shortcuts

7084054 opened the pull request:
<!-- Hey there. Thank you so much for improving NewPipe, and filling out the details. Having roughly the same layout helps everyone considerably :)-->

#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
<!-- While bullet points are the norm in this section, feel free to write free-form text instead of a list -->
- made changes to database (adding in new bookmark column)   
- changes to search order priority to prioritise shortcut searches 
- added onclick on left search icon to shortcut or Un shortcut search.
- tap on search history icon to use. 

#### Before/After Screenshots/Screen Record
<!-- If your PR changes the app's UI in any way, please include screenshots or a video showing exactly what changed, so that developers and users can pinpoint it easily. Delete this if it doesn't apply to your PR.-->
- Before:
![image showing search](https://github.com/7084054/NewPipe/blob/dev/.github/before.png?raw=true)
- After:
![image showing search shortcut after search history icon clicked on](https://github.com/7084054/NewPipe/blob/dev/.github/After.png?raw=true)
tapped on search history icon. 
#### Fixes the following issue(s)
<!-- Prefix issues with "Fixes" so that GitHub closes them when the PR is merged (note that each "Fixes #" should be in its own item). Also add any other relevant links. -->
- Fixes #7686

#### APK testing
<!-- Use a new, meaningfully named branch. The name is used as a suffix for the app ID to allow installing and testing multiple versions of NewPipe, e.g. "commentfix", if your PR implements a bugfix for comments. (No names like "patch-0" and "feature-1".)  -->
<!-- Remove the following line if you directly link the APK created by the CI pipeline. Directly linking is preferred if you need to let users test.-->
- Installed the APK on the emulator running Android 13.
- Verified the adding and removing of shortcuts on empty and non-empty strings.
- Ensured the bookmarks remain after opening and closing app.
- Confirmed no crashes or unexpected behaviour during testing.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

DehaanSolo reviewed (commented):
Good that you are clear about changes made to the databases. I can't see any of the other changes that you made though, can you point me to the other files?

7084054 commented:
> Good that you are clear about changes made to the databases. I can't see any of the other changes that you made though, can you point me to the other files?

Added file
- 10.json
Updated Files
- SearchHistoryDAO.java
- SearchHistoryEntry.kt
- AppDatabase.java
- Migrations.java
- SearchFragment.java
- SuggestionItem.java
- SuggestionListAdapter.java
- HistoryRecordManager.java
- NewPipeDatabase.java
- item_search_suggestion.xml
Updated test cases: 
- DatabaseMigrationTest.kt 
- HistoryRecordManagerTest.kt

DehaanSolo commented:
Thanks. I took a look at some.  I don't entirely understand what is going on but I left a comment or two. Im happy to go ahead with a proper PR
